File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int size=nums.length;
        if(size==0) return 0;
        int sum=0;
        int maxSum=Integer.MIN_VALUE;
        for(int i=0;i<size;++i){
            sum+=nums[i];
            maxSum=Math.max(maxSum,sum);
            if(sum<0) sum=0;
        }
        return maxSum;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n=nums.size();
        int res=INT_MIN;
        int sum=0;
        for(int p:nums){
            sum+=p;
            res=max(res,sum);
            if(sum<0) sum=0;
        }
        return res;

    }
};